1) Na podstawie zebranych danych oblicz podstawowe miary położenia i rozproszenia dla średniej oczekiwanej długości życia.

n = 20

poziom_ufności = 95% / 
odchylenie_standardowe_populacji = 9,09

srednia
mediana
dominanta
wariancja
kowariancja

In [2]:
Fertility_rate = [1.3,1.95,2.466,1.6,2.952,1.801,1.34,1.34,1.63,2.326,2.436,6.563,1.283,1.43,1.988,1.61,1.921,2.4,1.495,4.705]

Life_expectancy = [76.84878,81.40732,77.57895,74.16341,68.84907,73.88595,75.26829,76.26098,80.57244,65.46259,67.5482,55.02451,76.2799,
82.29024,69.80695,81.40112,82.19756,74.22683,80.12888,45.55095]

import numpy as np

n=20
lif_exp = np.array(Life_expectancy)

mean = lif_exp.sum()/n
print(f"średnia oczekiwanej długości życia w 20 krajach: {mean}")

lif_exp = sorted(Life_expectancy)
if len(lif_exp)%2 == 0:
    median = (lif_exp[(n//2)]+lif_exp[(n//2+1)])/2
else:
    median = lif_exp[((n//2)+1)]
        
print(f"mediana oczekiwanej długości życia w 20 krajach: {median}")

print("brak dominanty oczekiwanej długości życia w 20 krajach")

średnia oczekiwanej długości życia w 20 krajach: 73.23764600000001
mediana oczekiwanej długości życia w 20 krajach: 76.27044000000001
brak dominanty oczekiwanej długości życia w 20 krajach


In [3]:
suma = 0
for number in Life_expectancy:
    suma += (number-mean)**2
variance = suma/(n-1)
print(f"wariancja oczekiwanej długości życia w 20 krajach: {variance}")

std_dev = variance**(1/2)
print(f"odchylenie standardowe oczekiwanej długości życia w 20 krajach: {std_dev}")

wariancja oczekiwanej długości życia w 20 krajach: 88.19411239750947
odchylenie standardowe oczekiwanej długości życia w 20 krajach: 9.39117204599668


2) Następnie zmierz korelację pomiędzy obiema zmiennymi i opisz jej siłę i kierunek. Jeśli korelacja występuje, zastanów się, czy można przypuszczać, że jest ona przypadkowa, czy może ma jakieś uzasadnienie.

In [4]:
n=20
fertility = np.array(Fertility_rate)

mean_2 = fertility.sum()/n
print(f"średnia współczynnika dzietności w 20 krajach: {mean_2}")

suma = 0
for number in Fertility_rate:
    suma += (number-mean_2)**2
variance_2 = suma/(n-1)
print(f"wariancja współczynnika dzietności w 20 krajach: {variance_2}")

std_dev_2 = variance_2**(1/2)
print(f"odchylenie standardowe współczynnika dzietności w 20 krajach: {std_dev_2}")

średnia współczynnika dzietności w 20 krajach: 2.2268
wariancja współczynnika dzietności w 20 krajach: 1.6653558526315793
odchylenie standardowe współczynnika dzietności w 20 krajach: 1.2904866727834035


Obliczenie korelacji:

In [18]:
suma = 0
for i in range(0, n):
    suma += ((Life_expectancy[i] - mean) * (Fertility_rate[i] - mean_2))

covariance = suma / (n-1)        

corelation = covariance/(std_dev*std_dev_2)

print(f"korelacja obu zmiennych w 20 krajach: {corelation}")

korelacja obu zmiennych w 20 krajach: -0.8204930122649574


Uzyskany wynik na poziomie -0.82 wskazuje na silną ujemną korelację oczekiwanej długości życia oraz współczynnika dzietności. Wzrost jednej zmiennej obserwowany jest wraz ze spadkiem drugiej, może to wynikać z warunków życiowych ludności zamieszkających kraje rozwinięte i kraje rozwijające się.

3) Na koniec postaraj się oszacować średnią oczekiwaną długość życia dla całej populacji przy poziomie ufności 95%. 
Przyjmij, że odchylenie standardowe populacji wynosi 9,09.

P(-ta < t < ta) = 0.95

P(-ta < (mean-m)/std_dev < ta) = 0.95

P(-2.0930 < (73.237646-m)/9.09 < 2.0930) = 0.95

P(92.263 > m > 54.2123) = 0.95

Szacuje się, że średnia oczekiwana długość życia dla całej populacji przy poziomie ufności 95% wyniesie między 92, a 54 lata.